Source/FieldSolver/SpectralSolver/SpectralKSpace.cpp

```diff
@@ -149,7 +149,7 @@ SpectralKSpace::getSpectralShiftFactor( const DistributionMapping& dm,
             case ShiftType::TransformToCellCentered: sign = 1.;
         }
         const Complex I{0,1};
-        for (int i=0; i<k.size(); i++ ){
+        for (int i=0; i<static_cast<int>(k.size()); i++ ){
             shift[i] = exp( I*sign*k[i]*0.5_rt*dx[i_dim]);
```

Comment:
For those loops, you can actually set a constant `int` variable before the loop.
`std::vector::size()` is a pointer calculation, so you call `.size()` each loop iteration, which generally will hamper vectorization attempts by the compiler.
https://github.com/gcc-mirror/gcc/blob/releases/gcc-10.2.0/libstdc++-v3/include/bits/stl_vector.h#L915-L919
```suggestion
        int const num_k = k.size();
        for (int i=0; i<num_k; i++ ){
            shift[i] = exp( I*sign*k[i]*0.5_rt*dx[i_dim]);
```

Alternatively, and equally efficient, is a range-base for loop in modern C++:
```suggestion
        int i = 0;
        for (auto const& kv : k){
            shift[i] = exp( I*sign*kv*0.5_rt*dx[i_dim]);
            i++;
```
but you would have to carry the index if you need it anyway.

Reply:
Thanks for the suggestion! I have committed your second solution... sometimes I really feel the need of something like `enumerate` in C++

Reply:
I used the same solution for the other two loops in the same file.

Reply:
Thanks, yes enumerate in the std lib would be sweet.

I see some people implement such a thing, which we can do, too - but we might want to check the efficiency of the generated code for the usual compilers (gcc/clang/nvcc). Example: https://faouellet.github.io/enumerate/